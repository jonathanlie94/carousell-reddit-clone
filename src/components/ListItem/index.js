import React, { Component } from 'react';
import styled from 'styled-components';
import { theme } from 'ui';
import PropTypes from 'prop-types';

const Wrapper = styled.div`
  background-color: ${theme.colors.white};
  border-bottom: 1px solid ${theme.colors.grey};
  min-height: 4rem;
  padding: 16px;
`;

class ListItem extends Component {
  render() {
    return (
      <Wrapper className={this.props.className}>
        {this.props.children}
      </Wrapper>
    );
  }
}

ListItem.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
}

export default ListItem;
