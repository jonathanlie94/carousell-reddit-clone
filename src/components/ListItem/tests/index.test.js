import React from 'react';
import { shallow } from 'enzyme';
import ListItem from '../';

describe('<ListItem />', () => {
  it('should render children when passed in', () => {
    const renderedComponent = shallow(
      <ListItem>
        <div className="unique-class-name" />
      </ListItem>
    );
    expect(
      renderedComponent.contains(<div className="unique-class-name" />)
    ).toEqual(true);
  });
});
