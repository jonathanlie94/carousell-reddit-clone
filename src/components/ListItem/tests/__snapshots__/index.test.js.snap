// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> should match previous snapshot 1`] = `
.dyiVyr {
  background-color: #FFF;
  border-bottom: 1px solid #C8C8C8;
  min-height: 4rem;
  padding: 16px;
}

<ListItem>
  <styled.div>
    <div
      className="sc-bdVaJa dyiVyr"
    />
  </styled.div>
</ListItem>
`;
